ivy/functional/backends/jax/general.py

```diff
@@ -129,7 +152,7 @@ def gather(
     batch_dims: int = 0,
     out: Optional[JaxArray] = None,
 ) -> JaxArray:
-    axis = axis % len(params.shape)
+    axis = get_positive_axis_for_gather(axis, params.ndim)
     batch_dims = batch_dims % len(params.shape)
```

Comment:
As was discussed on #26624, you will need to make the same change to all backends and also change the default value of the `axis` argument to 0.